transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SQLiteEventStore.java

```diff
@@ -531,6 +538,161 @@ private static Encoding toEncoding(@Nullable String value) {
+    String query = "SELECT 1 FROM log_event_dropped" + " WHERE log_source = ? AND reason = ?";
+    String[] selectionArgs = new String[] {logSource, Integer.toString(reason.getNumber())};
+    return inTransaction(
+        db -> tryWithCursor(db.rawQuery(query, selectionArgs), cursor -> cursor.getCount() > 0));
+  }
+
+  /**
+   * Load full-populated ClientMetrics.
+   *
+   * @return the returned ClientMetrics is fully populated with persisted and non-persisted fields.
+   */
+  @Override
+  public ClientMetrics loadClientMetrics() {
+    long currentTime = wallClock.getTime();
+    ClientMetrics.Builder clientMetricsBuilder = ClientMetrics.newBuilder();
+    Map<String, List<LogEventDropped>> metricsMap = new HashMap<>();
+    String query = "SELECT log_source, reason, events_dropped_count FROM log_event_dropped";
+
+    return inTransaction(
+        db ->
+            tryWithCursor(
+                db.rawQuery(query, new String[] {}),
+                cursor -> {
+                  while (cursor.moveToNext()) {
+                    String logSource = cursor.getString(0);
+                    LogEventDropped.Reason reason = convertToReason(cursor.getInt(1));
+                    long eventsDroppedCount = cursor.getLong(2);
+                    if (!metricsMap.containsKey(logSource)) {
+                      metricsMap.put(logSource, new ArrayList<>());
+                    }
+                    metricsMap
+                        .get(logSource)
+                        .add(
+                            LogEventDropped.newBuilder()
+                                .setReason(reason)
+                                .setEventsDroppedCount(eventsDroppedCount)
+                                .build());
+                  }
+                  populateLogSourcesMetrics(clientMetricsBuilder, metricsMap);
+                  setTimeWindow(clientMetricsBuilder, currentTime);
+                  setGlobalMetrics(clientMetricsBuilder);
+                  clientMetricsBuilder.setAppNamespace(packageName.get());
+                  return clientMetricsBuilder.build();
+                }));
+  }
+
+  private void populateLogSourcesMetrics(
+      ClientMetrics.Builder clientMetricsBuilder, Map<String, List<LogEventDropped>> metricsMap) {
+    for (Map.Entry<String, List<LogEventDropped>> entry : metricsMap.entrySet()) {
+      clientMetricsBuilder.addLogSourceMetrics(
+          LogSourceMetrics.newBuilder()
+              .setLogSource(entry.getKey())
+              .setLogEventDroppedList(entry.getValue())
+              .build());
+    }
+  }
+
+  private void setTimeWindow(ClientMetrics.Builder clientMetricsBuilder, long currentTime) {
+    inTransaction(
+        db ->
+            tryWithCursor(
+                db.rawQuery(
+                    "SELECT last_metrics_upload_ms FROM global_log_event_state", new String[] {}),
+                cursor -> {
+                  while (cursor.moveToNext()) {
```

Comment:
Do we expect there to ever be more than 1 row? if not, should we add something like `limit 1`?